src/alert_rules/prometheus/kafka_metrics.rules

```diff
@@ -0,0 +1,249 @@
+  # ================
+  - alert: ISR Expands Rate
+    expr: max(rate(kafka_server_replicamanager_isrexpandspersec{juju_charm!=".*"}[5m])) by (instance) != 0
+    for: 1m
+    labels:
+      severity: warning
+    annotations:
+      summary: 'Broker {{ $labels.instance }} :: In Sync Replicas :: {{ $value }} ISR Expansion Rate.'
+      description: 'If a broker goes down, ISR for some of the partitions shrink. When that broker is up again, ISRs are expanded once the replicas are fully caught up. Other than that, the expected value for ISR expansion rate is 0. If ISR is expanding and shrinking frequently, adjust Allowed replica lag.'
+
+  - alert: ISR Shrinks Rate
+    expr: max(rate(kafka_server_replicamanager_isrshrinkspersec{juju_charm!=".*"}[5m])) by (instance) != 0
+    for: 1m
+    labels:
+      severity: warning
+    annotations:
+      summary: 'Broker {{ $labels.instance }} :: In Sync Replicas :: {{ $value }} ISR Shrink Rate.'
+      description: 'If a broker goes down, ISR for some of the partitions shrink. When that broker is up again, ISRs are expanded once the replicas are fully caught up. Other than that, the expected value for ISR shrink rate is 0. If ISR is expanding and shrinking frequently, adjust Allowed replica lag.'
+
+  # ================
+  # Leader Elections
+  # ================
+  - alert: Leader Election Rate
+    expr: max(rate(kafka_controller_controllerstats_leaderelectionrateandtimems{juju_charm!=".*",quantile=""}[2m]))by(instance) !=0
+    for: 60s
+    labels:
+      severity: critical
+    annotations:
+      summary: "Broker Kafka :: Leader Elections :: Number of disputed leader elections rate are {{ $value }}"
+      description: "Critical: Kafka number of disputed leader elections rate(!=0) on the instance {{ $labels.instance }} for more than 1 minutes"
+
+  - alert: Unclean Leader Election
+    expr: max(rate(kafka_controller_controllerstats_uncleanleaderelectionspersec{juju_charm!=".*"}[5m])) by (instance) != 0
+    for: 1m
+    labels:
+      severity: critical
+    annotations:
+      summary: 'Broker {{ $labels.instance }} :: Leader Elections :: {{ $value }} unclean leader elections.'
+      description: '{{ $value }} unclean partition leader elections in the cluster reported in the last 1 minute interval. When unclean leader election is held among out-of-sync replicas, there is a possibility of data loss if any messages were not synced prior to the loss of the former leader. So if the number of unclean elections is greater than 0, investigate broker logs to determine why leaders were re-elected, and look for WARN or ERROR messages. Consider setting the broker configuration parameter unclean.leader.election.enable to false so that a replica outside of the set of in-sync replicas is never elected leader.'
+
+  # ==============
+  # Consumer Level
+  # ==============
+  - alert: Records Lag Max
+    expr: sum(kafka_server_fetcherlagmetrics_consumerlag{juju_charm!=".*"}) by(instance, client_id) > 0
```

Comment:
I didn't get this variable to appear even after adding data-integrator

Reply:
I believe you need to add a consumer too (I have added that with the `kafka-test-app`, not the data-integrator)